src/semver.toit

```diff
@@ -1,195 +1,501 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+
+// Text code had a method for determining if a semver was valid.
+is-valid
+    input/any
+    --accept-missing-minor=false
+    --accept-missing-patch=false
+    --non-throwing=false
+    --accept-leading-zeros=false
+    --accept-v=false
+    -> bool:
+  // Normalize both sides to SemanticVersion
+  parsed-input := SemanticVersion.parse input --accept-missing-minor=accept-missing-minor --accept-missing-patch=accept-missing-patch --non-throwing=non-throwing --accept-leading-zeros=accept-leading-zeros --accept-v=accept-v
+  return (parsed-input != null)
+
+// Text code had a method for comparing two arbitrary versions.
+compare input-a/any input-b/any [--if-equal] -> int:
+  // Normalize both sides to SemanticVersion
+  a/SemanticVersion? := (input-a is SemanticVersion) ? input-a : SemanticVersion.parse input-a
+  b/SemanticVersion? := (input-b is SemanticVersion) ? input-b : SemanticVersion.parse input-b
+  if (a is not SemanticVersion) or (b is not SemanticVersion):
+    throw "compare: Unable to parse one (or both) inputs."
+  output:= a.compare-to b
+  if output == 0:
+    return if-equal.call
+  else:
+    return 0
 
-  return [semver, prerelease]
+// Text code had a method for executing a block, for two arbitrary versions.
+compare input-a/any input-b/any -> int:
+  return compare input-a input-b --if-equal=: 0
 
 /**
-Compares a dotted part of a semver string.
-
-This is used for major.minor.patch, but also for prerelease and build.
+A semantic versioning library.
 
-Generally, the rules are:
-- If one is an integer and the other is not, the integer is lower.
-- If both are integers compare them.
-- If both are strings compare them.
+See https://semver.org/ for details.
 */
-compare-dotted_ a/string b/string -> int:
-  a-parts := a.split "."
-  b-parts := b.split "."
-
-  max-parts := max a-parts.size b-parts.size
-  for i := 0; i < max-parts; i++:
-    // If a part is missing it is considered to be equal to 0.
-    a-part := i < a-parts.size ? a-parts[i] : "0"
-    b-part := i < b-parts.size ? b-parts[i] : "0"
-
-    a-not-int := false
-    a-part-int := int.parse a-part --if-error=:
-      a-not-int = true
-      -1
-
-    b-not-int := false
-    b-part-int := int.parse b-part --if-error=:
-      b-not-int = true
-      -1
-
-    // Semver requires major, minor and patch to be integers.
-    // If we get something else we simply compare the two strings without
-    // converting to a number.
-    if a-not-int and b-not-int:
-      comp := a-part.compare-to b-part
-      if comp != 0: return comp
-    // If one is an integer and the other is not, the integer is lower.
-    else if a-not-int:
+
+class SemanticVersion:
+  _major/int
+  _minor/int
+  _patch/int
+  _pre-releases/List
+  _build-numbers/List
```

Comment:
We don't use leading `_`.
Also just make them public. No need to go through getters.